creme/optim/nadam.py

```diff
@@ -0,0 +1,68 @@
+import collections
+import math
+
+from . import base
+
+
+__all__ = ['Nadam']
+
+
+class Nadam(base.Optimizer):
+    """Nadam optimizer.
+
+    Example:
+
+        ::
+
+            >>> from creme import linear_model
+            >>> from creme import metrics
+            >>> from creme import model_selection
+            >>> from creme import optim
+            >>> from creme import preprocessing
+            >>> from creme import stream
+            >>> from sklearn import datasets
+
+            >>> X_y = stream.iter_sklearn_dataset(
+            ...     dataset=datasets.load_breast_cancer(),
+            ...     shuffle=True,
+            ...     random_state=42
+            ... )
+            >>> optimizer = optim.Nadam()
+            >>> model = (
+            ...     preprocessing.StandardScaler() |
+            ...     linear_model.LogisticRegression(optimizer)
+            ... )
+            >>> metric = metrics.F1()
+
+            >>> model_selection.progressive_val_score(X_y, model, metric)
+            F1: 0.955307
+
+    References:
+        1. `Nadam: A combination of adam and nesterov <https://ruder.io/optimizing-gradient-descent/index.html#nadam>`_
+
+    """
+
+    def __init__(self, lr=0.1, beta_1=0.9, beta_2=0.999, eps=1e-8):
+        super().__init__(lr)
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.eps = eps
+        self.m = collections.defaultdict(float)
+        self.v = collections.defaultdict(float)
+
+    def _update_after_pred(self, w, g):
+
+        # Correct bias for `v`
+        lr = self.learning_rate * \
+            (1 - self.beta_2 ** (self.n_iterations + 1)) ** .5
+        # Correct bias for `m`
+        lr /= (1 - self.beta_1 ** (self.n_iterations + 1))
+
+        for i, gi in g.items():
+            self.m[i] = self.beta_1 * self.m[i] + (1 - self.beta_1) * gi
+            self.v[i] = self.beta_2 * self.v[i] + (1 - self.beta_2) * gi ** 2
+
+            w[i] -= lr * (self.beta_1*self.m[i] + (1-self.beta_1)*gi*(1-math.pow(
+                self.beta_1, self.n_iterations+1))) / (self.v[i] ** .5 + self.eps)
```

Comment:
The implementation is almost correct but I see two problems (correct me if I am wrong), the first one is here `gi*(1-math.pow(
                self.beta_1, self.n_iterations+1))`, it should be a division instead of a multiplication between `gi` et the `(1-math.pow(...)`

The second one is a little bit tricky but I will try to explain you with some math.

Reply:
The issue is because you do the bias correction in the learning rate which is usaually not a problem but in the case of NADAM, it is:

![NADAM](https://user-images.githubusercontent.com/32324822/72790468-f172fe80-3c35-11ea-97f6-4b42455f2230.png)

So you should do the bias correction at each step.

Reply:
> The implementation is almost correct but I see two problems (correct me if I am wrong), the first one is here `gi*(1-math.pow( self.beta_1, self.n_iterations+1))`, it should be a division instead of a multiplication between `gi` et the `(1-math.pow(...)`
> 
Thank you @VaysseRobin for this, it was a typo and lucky it worked well even with this. I changed it now

Reply:
> The issue is because you integrate bias correction in the learning rate which is usaually not a problem but in the case of NADAM, it is:
> 
> ![NADAM](https://user-images.githubusercontent.com/32324822/72790468-f172fe80-3c35-11ea-97f6-4b42455f2230.png)
> 
> So you should do the bias correction at each step.

@VaysseRobin I changed it now. Thankyou :)